pkg/client/client_test.go

```diff
@@ -1870,13 +1869,15 @@ func TestCreateOrUpdatePrometheus(t *testing.T) {
 					},
 				},
 			},
+			shouldUpdate: true,
 		},
 		{
 			description: "apply no value to the existing prometheus",
 			prometheus: &monv1.Prometheus{
 				TypeMeta:   initialPrometheus.TypeMeta,
 				ObjectMeta: initialPrometheus.ObjectMeta,
 			},
+			shouldUpdate: true,
```

Comment:
With the no defaulting approach, there won't be a difference between default and non-default applications, as both will trigger updates as compared to just the latter earlier, at the benefit of not maintaining hardcoded default values in CMO. LMK if I should revert this or if this sounds good.